Src/main.c

```diff
@@ -255,10 +258,19 @@ int main(void)
 		  else
 			  blinkTime = 500;
 
+	  }
+
+	  static const uint32_t baroSampleInterval = 250;
+	  static uint32_t lastBaroSampleTime = 0;
 
+	  if(currentTime - lastBaroSampleTime > baroSampleInterval)
+	  {
+		  lastBaroSampleTime=currentTime;
```

Comment:
If you implement the changes I mentioned above, you should be able to reduce this significantly. 15ms should guarantee that there will always be a sample waiting if you configure the ODR for 75Hz.
Also, I would only set lastBaroSampleTime when a successful read happens (when the status register indicates data is actually available, and a read is performed).